modules/openapi-generator/src/main/resources/dart2/api_client.mustache

```diff
@@ -1,11 +1,13 @@
 {{>header}}
 {{>part_of}}
-class ApiClient {
-  ApiClient({this.basePath = '{{{basePath}}}', this.authentication});
+class ApiClient<T extends Authentication> {
+  ApiClient({this.basePath = '{{{basePath}}}', this.authentication,});
 
   final String basePath;
+  final T? authentication;
 
   var _client = Client();
+  final _defaultHeaderMap = <String, String>{};
```

Comment:
if you want to expose this, shouldn't just a simple rename to
```dart
final defaultHeaderMap = <String, String>{};
``` 
work ?

Reply:
I just moved properties around so private ones huddle together, so no logic change in this pr.